lib/src/rules/unnecessary_string_interpolations.dart

```diff
@@ -57,7 +58,9 @@ class _Visitor extends SimpleAstVisitor<void> {
       var end = node.elements[2] as InterpolationString;
       if (start.value.isEmpty && end.value.isEmpty) {
         var staticType = interpolation.expression.staticType;
-        if (staticType != null && staticType.isDartCoreString) {
+        if (staticType != null &&
+            staticType.isDartCoreString &&
+            staticType.nullabilitySuffix != NullabilitySuffix.question) {
```

Comment:
What do we want the behavior to be if the type is `*`, meaning that it comes from a legacy library? Should we continue to lint it even though it might be nullable?

Reply:
My gut is that we do want to lint it but I'm really not sure.  It'll be interesting to see if this comes up in practice?